app/lib/spam.js

```diff
@@ -0,0 +1,27 @@
+import { getConfig } from '#app/config'
+import { isNonEmptyArray } from '#lib/boolean_tests'
+
+let suspectKeywordsPattern
+
+async function initSuspectKeywordsPattern () {
+  if (suspectKeywordsPattern === undefined) {
+    const config = await getConfig()
+    if (isNonEmptyArray(config.spam?.suspectKeywords)) {
+      const { suspectKeywords } = config.spam
+      suspectKeywordsPattern = new RegExp(`(${suspectKeywords.join('|')})`, 'i')
+    } else {
+      suspectKeywordsPattern = null
+    }
+  }
+}
+
+export async function looksLikeSpam (str) {
+  await initSuspectKeywordsPattern()
+  if (suspectKeywordsPattern) {
+    return urlPattern.test(str) && suspectKeywordsPattern.test(str)
+  } else {
+    return false
+  }
+}
+
+const urlPattern = /(http|www\.|\w+\.\w+\/)/i
```

Comment:
An if/else from my initial implementation could indeed be dropped, with the help of the `?.` operator
```js
export async function looksLikeSpam (str) {
  await initSuspectKeywordsPattern()
  return suspectKeywordsPattern?.test(str) && urlPattern.test(str)
}
```
but I would prefer an implementation that doesn't unnecessarily redefine `suspectKeywordsPattern` at every call

Reply:
indeed, moving `let suspectKeywordsPattern` below `let config` does the trick, no? Then `initSuspectKeywordsPattern` is called only once since `if (!config)` prevents requesting config at every call no?

Reply:
I find both our implementation hard to read. What do you think of an implementation relying on an `initialized` boolean to explicit what's happening there? f11895842

Reply:
yep